src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -38,17 +38,168 @@ namespace System.Text.Json
+        /// Copies the current JSON token value from the source, unescaped, and transcoded as a UTF-16 char buffer.
+        /// </summary>
+        /// <param name="destination">A buffer to write the transcoded UTF-16 characters into.</param>
+        /// <returns>The number of characters written to <paramref name="destination"/>.</returns>
+        /// <remarks>Unlike <see cref="GetString"/>, this method does not support <see cref="JsonTokenType.Null"/>.</remarks>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of the JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>.
+        /// <seealso cref="TokenType" />
+        /// It will also throw when the JSON string contains invalid UTF-8 bytes, or invalid UTF-16 surrogates.
+        /// </exception>
+        /// <exception cref="ArgumentException">Thrown if the destination buffer is too small to hold the unescaped value.</exception>
+        public readonly int CopyString(Span<char> destination)
+        {
+            if (_tokenType is not (JsonTokenType.String or JsonTokenType.PropertyName))
+            {
+                ThrowHelper.ThrowInvalidOperationException_ExpectedString(_tokenType);
+            }
+
+            ReadOnlySpan<byte> unescapedSource = stackalloc byte[0];
+            byte[]? rentedBuffer = null;
+            int valueLength;
+
+            if (ValueIsEscaped)
+            {
+                valueLength = ValueLength;
+
+                Span<byte> unescapedBuffer = valueLength <= JsonConstants.StackallocByteThreshold ?
+                    stackalloc byte[JsonConstants.StackallocByteThreshold] :
+                    (rentedBuffer = ArrayPool<byte>.Shared.Rent(valueLength));
+
+                int bytesWritten = CopyStringWithEscaping(unescapedBuffer, isUserProvidedBuffer: false);
+                unescapedSource = unescapedBuffer.Slice(0, bytesWritten);
+            }
+            else
+            {
+                if (HasValueSequence)
+                {
+                    valueLength = checked((int)ValueSequence.Length);
+
+                    Span<byte> intermediate = valueLength <= JsonConstants.StackallocByteThreshold ?
+                        stackalloc byte[JsonConstants.StackallocByteThreshold] :
+                        (rentedBuffer = ArrayPool<byte>.Shared.Rent(valueLength));
+
+                    ValueSequence.CopyTo(intermediate);
+                    unescapedSource = intermediate.Slice(0, valueLength);
+                }
+                else
+                {
+                    unescapedSource = ValueSpan;
+                }
+            }
+
+            int charsWritten = JsonReaderHelper.TranscodeHelperWithCleanupOnFailure(unescapedSource, destination);
+
+            if (rentedBuffer != null)
+            {
+                new Span<byte>(rentedBuffer, 0, unescapedSource.Length).Clear();
```

Comment:
Do we need to clear the buffer before returning it?  Do we consistently do that elsewhere in System.Text.Json?  Outside of crypto we generally haven't elsewhere.

Reply:
Yes, it's fairly common practice in the STJ codebase.

Reply:
Why?

Reply:
It's a decision that predates my STJ tenure, but I would guess it's to avoid leaking data to other users of the shared buffer pool? @steveharter might have more context.